### PR TITLE
DP-1.4: Handle differences in QoS queue names/numbers across vendors

### PR DESCRIPTION
**Pull request by sgirishj**
Handle differences in QoS queue names/numbers across vendors.

### CODE REVIEW

**Comment by xw-g** [comment]
@Hobbydos can you help comment for Nokia? Thanks.

**Review by sezhang2 — CHANGES_REQUESTED** [review]
Please wait for https://github.com/openconfig/featureprofiles/pull/1100 to be merged.

**Review by sachendras — APPROVED** [review]

**Comment by sezhang2** [comment]
What is the reason to change the queue names?

**Comment by sgirishj** [comment]
> 

Queues are identified using numeric values 0-7 in Juniper's case. Added switch-case construct where such nomenclature is needed. For the other vendors, names are left to what they were in the script previously.

**Comment by sgirishj** [comment]
Closed by mistake. Reopening.

**Comment by sezhang2** [comment]
> 

Do you have a bug to track this limitation?

**Comment by sgirishj** [comment]
> > 
> 
> Do you have a bug to track this limitation?

No. Is there a requirement?

**Comment by sezhang2** [comment]
> > > 
> > 
> > 
> > Do you have a bug to track this limitation?
> 
> No. Is there a requirement?

It is a workaround to use queue number. The requirement is to use queue name, so we need to track this workaround.

From https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-qos.html#qos-interfaces-interface-output-queues-queue-name

**/qos/queues/queue/config/
name
description:
User-defined name of the queue

nodetype: leaf (rw)
type: string**

File: feature/experimental/qos/ate_tests/qos_output_queue_counters_test/qos_output_queue_counters_test.go [hunk 1]
```diff
@@ -388,38 +388,58 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
 		gnmi.Replace(t, dut, gnmi.OC().Qos().Config(), q)
 	}
 
+	var q_BE1, q_BE0, q_AF1, q_AF2, q_AF3, q_AF4, q_NC1 string
```
Comment (greg-dennis) [inline feature/experimental/qos/ate_tests/qos_output_queue_counters_test/qos_output_queue_counters_test.go:391]: underscores are not allowed in go variable name style. I'd probably create an inline type here:

	type qosVals struct {
		be0, be1, af1, af2, af3, af4, nc1 string
	}
	qos := qosVals {
		be0: "BE0",
		be1: "BE1",
		af1: "AF1",
		af2: "AF2",
		af3: "AF3",
		af4: "AF4",
		nc1: "NC1",
	}
	if dut.Vendor() == ondatra.JUNIPER {
		qos = qosVals {
			be0: "7",
			be1: "0",
			af1: "4",
			af2: "1",
			af3: "5",
			af4: "2",
			nc1: "3",
		}
	}

File: feature/experimental/qos/ate_tests/qos_output_queue_counters_test/qos_output_queue_counters_test.go [hunk 2]
```diff
@@ -388,38 +388,58 @@ func ConfigureQoS(t *testing.T, dut *ondatra.DUTDevice) {
...
 	t.Logf("Create qos forwarding groups config")
 	forwardingGroups := []struct {
 		desc         string
 		queueName    string
 		targetGrpoup string
 	}{{
 		desc:         "forwarding-group-BE1",
-		queueName:    "BE1",
+		queueName:    q_BE1,
 		targetGrpoup: "target-group-BE1",
```
Comment (greg-dennis) [inline feature/experimental/qos/ate_tests/qos_output_queue_counters_test/qos_output_queue_counters_test.go:419]: please fix the misspelling "targetGroup"

**Review by greg-dennis — CHANGES_REQUESTED** [review]

**Review by sezhang2 — DISMISSED** [review]
Per the discussion, it is a workaround. It should be fixed via b/269154477.

**Review by greg-dennis — APPROVED** [review]

**Pull request merged**